src/screenshot.cpp

```diff
@@ -682,7 +174,7 @@ static const char *MakeScreenshotName(const char *default_fn, const char *ext, b
 /** Make a screenshot of the current screen. */
 static bool MakeSmallScreenshot(bool crashlog)
 {
-	return _cur_screenshot_format->proc(MakeScreenshotName(SCREENSHOT_NAME, _cur_screenshot_format->extension, crashlog), CurrentScreenCallback, nullptr, _screen.width, _screen.height,
+	return _cur_screenshot_format->MakeImage(MakeScreenshotName(SCREENSHOT_NAME, _cur_screenshot_format->GetName(), crashlog), CurrentScreenCallback, nullptr, _screen.width, _screen.height,
```

Comment:
Since `_cur_screenshot_format` is now initialised to `nullptr`, won't this segfault if a crash happens before `InitializeScreenshotFormats` is called? 
I know it might've been able if it crashed before `_cur_screenshot_format` got initialized before, but that's a much smaller bit of code I reckon.
```suggestion
	return _cur_screenshot_format != nullptr && _cur_screenshot_format->MakeImage(MakeScreenshotName(SCREENSHOT_NAME, _cur_screenshot_format->GetName(), crashlog), CurrentScreenCallback, nullptr, _screen.width, _screen.height,
```

Though it looks like there are other things that cause the crashlog to crash in early startup. But that's not for this PR.

Reply:
Ah, if it tries to screenshot before it's initialised for whatever reason. Hmm. Likely unable to do so anyway, but yeah.

Reply:
I might not been as clear as I wanted to be, or I thought to be.

My point is that `_cur_screenshot_format` will be `nullptr` and that when a crash happens before `InitializeScreenshotFormats` is called, it will crash the crash log handler. Checking for `nullptr` in the only screenshot function that creates the crash log seems like an easy fix.

I do realise that there are other cases that will crash in the crash log handler, as this already happens. I do think that we should solve those cases, but definitely not in this PR.